src/Compilers/CSharp/Portable/CodeGen/EmitStatement.cs

```diff
@@ -1482,6 +1482,13 @@ private string GetLocalDebugName(ILocalSymbolInternal local, out LocalDebugId lo
                 return null;
             }
 
+            // Skip naming locals generated by instrumentation as they might not have a local syntax offset.
+            // Locals generated by instrumentation might exist in methods which do not contain a body (auto property initializers).
+            if (local.SynthesizedKind == SynthesizedLocalKind.InstrumentationPayload)
+            {
+                return null;
```

Comment:
Could you provide more details about what is failing and where and why if we don't do this?